dwds/web/run_main.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:html';
+
+/// Creates a script that will run properly when strict CSP is enforced.
+///
+/// More specifically, the script has the correct `nonce` value set.
+final ScriptElement Function() _createScript = (() {
+  final nonce = _getNonce();
+  if (nonce == null) return () => ScriptElement();
+
+  return () => ScriptElement()..setAttribute('nonce', nonce);
+})();
+
+// According to the CSP3 spec a nonce must be a valid base64 string.
+final _noncePattern = RegExp('^[\\w+\/_-]+[=]{0,2}\$');
+
+/// Returns CSP nonce, if set for any script tag.
+String _getNonce() {
```

Comment:
[nit]

```suggestion
String _nonce() {
```

Or maybe `_findNonce`?

https://dart.dev/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get